### PR TITLE
Properly report when command reached timeout

### PR DESCRIPTION
**Pull request by mrnugget**
Before this we just reported that the process was killed, but not that the kill signal was triggered by the timeout.

### CODE REVIEW

File: cmd/src/actions_exec_backend_runner.go [hunk 1]
```diff
@@ -112,6 +112,16 @@ func (e *errTimeoutReached) Error() string {
 	return fmt.Sprintf("Timeout reached. Execution took longer than %s.", e.timeout)
 }
 
+func reachedTimeout(cmdCtx context.Context, err error) bool {
+	if ee, ok := errors.Cause(err).(*exec.ExitError); ok {
+		if ee.String() == "signal: killed" && cmdCtx.Err() == context.DeadlineExceeded {
```
Comment (ryanslade) [inline cmd/src/actions_exec_backend_runner.go]: I think there's a chance of returning a false positive here.

We could have received a kill signal and THEN the timeout happened. We're only checking the value of `cmdCtx.Err()` after we already know `err` isn't nil

Comment (mrnugget) [reply cmd/src/actions_exec_backend_runner.go]: Yeah, I know. But is there a reliable way of checking that the timeout is what caused the process to be killed? Even if I turn the check around and check `cmdCtx.Err()` as the first thing, there's still a chance that the process was killed by a user-signal and right after we ran into the timeout, right?

Comment (ryanslade) [reply cmd/src/actions_exec_backend_runner.go]: I also couldn't think of a reliable way to check but wanted to point it out in case it was a deal breaker

Comment (mrnugget) [reply cmd/src/actions_exec_backend_runner.go]: Yep, appreciated. I think this is better than not reporting anything, though.

**Pull request merged**